src/vm/appdomain.cpp

```diff
@@ -6788,11 +6788,60 @@ DomainAssembly *AppDomain::LoadDomainAssemblyInternal(AssemblySpec* pIdentity,
     
     if (result == NULL)
     {
+#if defined(FEATURE_HOST_ASSEMBLY_RESOLVER) && defined(FEATURE_COLLECTIBLE_ALC) && !defined(CROSSGEN_COMPILE)
+        BOOL fIsCollectible = FALSE;
+        SafeComHolder<IAssemblyLoadContext> pAssemblyLoadContext = NULL;
+
+        ICLRPrivBinder *pFileBinder = pFile->GetBindingContext();
+        if (pFileBinder != NULL)
+        {
+            ICLRPrivBinder *pBinder = reinterpret_cast<BINDER_SPACE::Assembly *>(pFileBinder)->GetBinder();
+
+            // Assemblies loaded with AssemblyLoadContext need to use a different LoaderAllocator if
+            // marked as collectible
+            if (SUCCEEDED(pBinder->QueryInterface<IAssemblyLoadContext>(&pAssemblyLoadContext)))
+            {
+                pAssemblyLoadContext->GetIsCollectible(&fIsCollectible);
+            }
+        }
+
+        AssemblyLoaderAllocator *pLoaderAllocatorOverride = NULL;
+
+        if (fIsCollectible && pAssemblyLoadContext != NULL)
```

Comment:
pAssemblyLoadContext is always non-NULL when fIsCollectible is TRUE - please simplify the condition to only check for fIsCollectible to be TRUE.